hw22-cache/src/main/java/ru/otus/crm/service/DbServiceClientImpl.java

```diff
@@ -0,0 +1,126 @@
+package ru.otus.crm.service;
+
+import java.util.List;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import ru.otus.cachehw.MyCache;
+import ru.otus.core.repository.DataTemplate;
+import ru.otus.core.sessionmanager.TransactionManager;
+import ru.otus.crm.model.Client;
+
+public class DbServiceClientImpl implements DBServiceClient {
+    private static final Logger log = LoggerFactory.getLogger(DbServiceClientImpl.class);
+
+    private final DataTemplate<Client> clientDataTemplate;
+    private final TransactionManager transactionManager;
+    private final MyCache<String, Object> clientCache;
+
+    public DbServiceClientImpl(
+            TransactionManager transactionManager,
+            DataTemplate<Client> clientDataTemplate,
+            MyCache<String, Object> clientCache) {
```

Comment:
Лучше тип объявлять интерфейсом.
```suggestion
    private final HwCache<String, Object> clientCache;

    public DbServiceClientImpl(
            TransactionManager transactionManager,
            DataTemplate<Client> clientDataTemplate,
            HwCache<String, Object> clientCache) {
```